AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -426,3 +409,31 @@ extension RemoteSyncEngine: AuthModeStrategyDelegate {
         return auth?.getCurrentUser() != nil
     }
 }
+
+@available(iOS 13.0, *)
+extension RemoteSyncEngine: Resettable {
+    func reset(onComplete: @escaping BasicClosure) {
+        let group = DispatchGroup()
+
+        let mirror = Mirror(reflecting: self)
+        for child in mirror.children {
+            guard child.label != "api" else {
+                log.verbose("Not resetting API category from RemoteSyncEngine")
+                continue
+            }
```

Comment:
what's the reason for excluding `API` but not the `auth` category?

Reply:
Funnily enough, I'm doing that in another Resettable tranche coming shortly :)